Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -728,6 +728,9 @@ void AODProducerWorkflowDPL::fillMCParticlesTable(o2::steer::MCKinematicsReader&
       if (daughters[0] > daughters[1]) {
         std::swap(daughters[0], daughters[1]);
       }
+      if (daughters[0] < 0 && daughters[1] >= 0) {
+        LOG(fatal) << "Discordant daughters indices (" << daughters[0] << "," << daughters[1] << "): the AO2D would be nonconforming to the data model, aborting.";
+      }
```

Comment:
so just to understand: if daughter index 0 is negative (== no daughter) we should not have a daughter in index 1 (second daughter)?

It appears to me that the 2 lines above will already cause such misbehaviour. We seem to swap the index whenever daughters[0] > daughters[1] which happens in particular when daughters[1] == -1. @shahor02 : Can you comment on this?